tools/caps/main.go

```diff
@@ -0,0 +1,214 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/GeoNet/delta/internal/metadb"
+	"github.com/GeoNet/delta/resp"
+)
+
+const contents = `###
+### Delivered by puppet
+###
+capslink:sl4caps
+`
+
+type Station struct {
+	Network string
+	Code    string
+}
+
+func (s Station) Key() string {
+	return fmt.Sprintf("%s_%s", strings.ToUpper(s.Network), strings.ToUpper(s.Code))
+}
+
+func (s Station) Path() string {
+	return fmt.Sprintf("station_%s", s.Key())
+}
+
+func (s Station) Store(path string) error {
+	if err := os.MkdirAll(filepath.Dir(path), 0755); err != nil {
+		return err
+	}
+	if err := ioutil.WriteFile(path, []byte(contents), 0644); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s Station) Output(base string) error {
+	if err := s.Store(filepath.Join(base, s.Path())); err != nil {
+		return err
+	}
+	return nil
+}
+
+func main() {
+
+	var base string
+	flag.StringVar(&base, "base", "../..", "delta base files")
+
+	var output string
+	flag.StringVar(&output, "output", "keys", "output caps configuration directory")
+
+	var networks string
+	//TODO: this could be managed via configuration elsewhere.
+	flag.StringVar(&networks, "networks", "AK,CB,CH,EC,FI,HB,KI,MN,NX,NZ,OT,RT,SC,SI,SM,SP,SX,TP,TR,WL", "comma separated list of networks to use")
+
+	var exclude string
+	flag.StringVar(&exclude, "exclude", "", "comma separated list of stations to skip (either NN_SSSS, or simply SSSS)")
+
+	var include string
+	flag.StringVar(&include, "include", "", "comma separated list of external stations to add (requires NN_SSSS)")
+
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "\n")
+		fmt.Fprintf(os.Stderr, "Build a set of caps key files\n")
+		fmt.Fprintf(os.Stderr, "\n")
+		fmt.Fprintf(os.Stderr, "Usage:\n")
+		fmt.Fprintf(os.Stderr, "\n")
+		fmt.Fprintf(os.Stderr, "  %s [options]\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "\n")
+		fmt.Fprintf(os.Stderr, "Options:\n")
+		fmt.Fprintf(os.Stderr, "\n")
+		flag.PrintDefaults()
+		fmt.Fprintf(os.Stderr, "\n")
+	}
+
+	flag.Parse()
+
+	check := make(map[string]interface{})
+	for _, n := range strings.Split(networks, ",") {
+		check[strings.TrimSpace(strings.ToUpper(n))] = true
+	}
+
+	var unmatch []string
+	for _, p := range strings.Split(exclude, ",") {
+		if s := func() string {
```

Comment:
The anonymous function strikes again. There's a lot going on here, if a function helps simplify it break it out into a named function; and maybe with some tests; otherwise simplifying it to more linear execution would help a lot.